library/core/src/option.rs

```diff
@@ -1037,7 +1037,7 @@ impl<T> Option<T> {
         match self {
             Some(val) => val,
             // SAFETY: the safety contract must be upheld by the caller.
-            None => unsafe { hint::unreachable_unchecked() },
+            None => unsafe { mem::MaybeUninit::uninit().assume_init() },
```

Comment:
is this really better? that's disappointing.. how about using offset_of and accessing the value with directly with a pointer offset, entirely removing the condition?

Reply:
I don't know if it's better in general - I just know it's better in at least one case. I think it's worth running a benchmark for it at least though. (which I can't do 🥺)

TIL about `offset_of`. `uninit` seems to be chucked pretty readily, but `offset_of` might be better in low opt level builds etc.

Reply:
@Nilstrieb It doesn't seem to have the desired effect:
https://godbolt.org/z/z5fv4as4s

@alion02 Please make your new libcore compile.